frappe/translate.py

```diff
@@ -267,19 +267,23 @@ def get_full_dict(lang):
 		return {}
 
 	# found in local, return!
-	if getattr(frappe.local, "lang_full_dict", None) is not None:
-		return frappe.local.lang_full_dict
+	if (lang_full_dict := getattr(frappe.local, "lang_full_dict", None)) is not None:
+		return lang_full_dict
 
-	frappe.local.lang_full_dict = load_lang(lang)
+	lang_full_dict = load_lang(lang)
 
 	try:
 		# get user specific translation data
-		user_translations = get_user_translations(lang)
-		frappe.local.lang_full_dict.update(user_translations)
+		if user_translations := get_user_translations(lang):
+			# copy to avoid cache mutation
+			lang_full_dict = lang_full_dict.copy()
```

Comment:
The original memory location of `lang_full_dict` is still being referenced by `frappe.local.cache["lang_full_dict"]`. You can analyze the function `load_lang` to understand that. Since we don't want to mutate the original cached object when updating, I created a new shallow copy.